ui/src/alerting/selectors/index.ts

```diff
@@ -14,3 +15,17 @@ export const getResourceIDs = (
     {}
   )
 }
+
+interface HasName {
+  name?: string
+}
+
+export const getResourceList = <T extends HasName>(
```

Comment:
My preference would be to repeat this code for each resource type so that it can be typechecked.

Reply:
With the above we get yelled at if the generic does not have a `name` field.